distributed/tests/py3_test_asyncio.py

```diff
@@ -0,0 +1,199 @@
+
+        L4 = c.map(add, range(3), range(4))
+        results = await c.gather(L4)
+        assert results == list(map(add, range(3), range(4)))
+
+        def f(x, y=10):
+            return x + y
+
+        L5 = c.map(f, range(5), y=5)
+        results = await c.gather(L5)
+        assert results == list(range(5, 10))
+
+        y = c.submit(f, 10)
+        L6 = c.map(f, range(5), y=y)
+        results = await c.gather(L6)
+        assert results == list(range(20, 25))
+
+
+async def test_asyncio_gather(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        x = c.submit(inc, 10)
+        y = c.submit(inc, x)
+
+        result = await c.gather(x)
+        assert result == 11
+        result = await c.gather([x])
+        assert result == [11]
+        result = await c.gather({'x': x, 'y': [y]})
+        assert result == {'x': 11, 'y': [12]}
+
+
+async def test_asyncio_get(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        result = await c.get({'x': (inc, 1)}, 'x')
+        assert result == 2
+
+        result = await c.get({'x': (inc, 1)}, ['x'])
+        assert result == [2]
+
+        result = await c.get({}, [])
+        assert result == []
+
+        result = await c.get({('x', 1): (inc, 1), ('x', 2): (inc, ('x', 1))},
+                              ('x', 2))
+        assert result == 3
+
+
+async def test_asyncio_exceptions(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        x = c.submit(div, 1, 2)
+        result = await x.result()
+        assert result == 1 / 2
+
+        x = c.submit(div, 1, 0)
+        with pytest.raises(ZeroDivisionError):
+            result = await x.result()
+
+        x = c.submit(div, 10, 2)  # continues to operate
+        result = await x.result()
+        assert result == 10 / 2
+
+
+async def test_asyncio_channels(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        x = c.channel('x')
+        y = c.channel('y')
+
+        assert len(x) == 0
+
+        while set(c.extensions['channels'].channels) != {'x', 'y'}:
+            await asyncio.sleep(0.01)
+
+        xx = c.channel('x')
+        yy = c.channel('y')
+
+        assert len(x) == 0
+
+        await asyncio.sleep(0.1)
+        assert set(c.extensions['channels'].channels) == {'x', 'y'}
+
+        future = c.submit(inc, 1)
+
+        x.append(future)
+
+        while not x.data:
+            await asyncio.sleep(0.01)
+
+        assert len(x) == 1
+
+        assert xx.data[0].key == future.key
+
+        xxx = c.channel('x')
+        while not xxx.data:
+            await asyncio.sleep(0.01)
+
+        assert xxx.data[0].key == future.key
+
+        assert 'x' in repr(x)
+        assert '1' in repr(x)
+
+
+async def test_asyncio_exception_on_exception(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        x = c.submit(lambda: 1 / 0)
+        y = c.submit(inc, x)
+
+        with pytest.raises(ZeroDivisionError):
+            await y.result()
```

Comment:
I have a slight preference to replace `await y.result()` with `await y` in most cases.  This seems like a more native syntax and people in the future will probably copy the style within these tests.